process/management/commands/finisher.py

```diff
@@ -40,6 +42,13 @@ def process(self, connection, channel, delivery_tag, body):
                     collection.completed_at = Now()
 
                     collection.save()
+
+                    # complete upgraded collection as well
+                    upgraded_collection = collection.get_upgraded_collection()
+                    if upgraded_collection:
+                        upgraded_collection.completed_at = Now()
+                        upgraded_collection.save()
+
```

Comment:
Because upgrade is running at the same phase as store of the original collection. Once data loaded from the filesystem, they are stored to both collections in the same transaction.